src/app/patient-dashboard/common/formentry/patient-transfer.service.ts

```diff
@@ -0,0 +1,209 @@
+          // all active programs should be stopped
+          _.merge(queryParams, {
+            stop: (this.componentRef.programClass).toUpperCase(),
+            notice: 'other'
+          });
+        }
+
+        // patient care status in transfer out form is ampath
+        if (_.first(patientCareStatus).control.value === 'a89c2e5c-1350-11df-a1f1-0026b9348838') {
+          _.merge(queryParams, {
+            change: (this.componentRef.programClass).toUpperCase(),
+            notice: 'location'
+          });
+        }
+      }
+      // MCH/PMTCT concept
+      // Transfer to MNCH for PMTCT care concept
+      if (_.includes([
+        '1f09e809-8ea3-45e6-a71f-16e6a0d72390',
+        'a8a17d80-1350-11df-a1f1-0026b9348838'], _.first(patientCareStatus).control.value)) {
+
+        // PMTCT programUuid
+        _.merge(queryParams, {
+          program: '781d897a-1359-11df-a1f1-0026b9348838',
+          notice: 'pmtct'
+        });
+      }
+      if (this.loadTransferOutForm()) {
+        this.transferState.next({
+          transfer: true,
+          loadTransferOutForm: true,
+        });
+      } else {
+        this.transferState.next({transfer: true, params: queryParams});
+      }
+    }
+    if (referralQuestion.length > 0) {
+      // Enhanced adherence HIV Program
+      if (_.includes(_.first(referralQuestion).control.value, 'a9431295-9862-405b-b694-534f093ca0ad')) {
+        // Enhanced adherence HIV Program
+        _.merge(queryParams, {
+          program: 'c4246ff0-b081-460c-bcc5-b0678012659e',
+          notice: 'adherence'
+        });
+        const location: any = this.componentRef.userDefaultPropertiesService.getCurrentUserDefaultLocationObject();
+        localStorage.setItem('transferLocation', location.uuid);
+      }
+      // HIV Differentiated Program
+      if (_.includes(_.first(referralQuestion).control.value, '7c6f0599-3e3e-4f42-87a2-2ce66f1e96d0')) {
+        // HIV Differentiated Program
+        _.merge(queryParams, {
+          program: '334c9e98-173f-4454-a8ce-f80b20b7fdf0',
+          notice: 'dc'
+        });
+        const location: any = this.componentRef.userDefaultPropertiesService.getCurrentUserDefaultLocationObject();
+        localStorage.setItem('transferLocation', location.uuid);
+      }
+    }
+    return this.transferState;
+  }
+
+  public prefillTransferOptions(): void {
+
+    const careStatus = this.searchNodeByQuestionId('careStatus');
+    if (careStatus.length > 0) {
+      // some delay just to ensure the question has been rendered then simulate a click before setting value.
+      setTimeout(() => {
+        const element: HTMLElement = document.getElementById('careStatusid') as HTMLElement;
+        element.click();
+        careStatus[0].control.setValue(localStorage.getItem('careStatus'));
+      }, 100);
+
+      // give some time for the hidden options to be shown
+      setTimeout(() => {
```

Comment:
It it possible to avoid this setTimeout()?

Reply:
Please see my comments